data/tutorials/en/000_up_and_running.md

```diff
@@ -123,57 +123,54 @@ OCaml compilation. We installed it with opam, above.
 
 ## A first project
 
-Let's begin the simplest project with Dune and OCaml. We create a new directory
-and ask `dune` to initialise a new project:
+Let's begin the simplest project with Dune and OCaml. We ask `dune` to
+initialize a new project, and then change into the created directory:
 
 ```
-$ mkdir helloworld
-$ cd helloworld/
-$ dune init exe helloworld
-Success: initialized executable component named helloworld
+$ dune init proj helloworld
+Success: initialized project component named helloworld
+$ cd helloworld
 ```
 
-Building our program is as simple as typing `dune build`:
+We can build our program with `dune build`:
 
 ```
 $ dune build
-Info: Creating file dune-project with this contents:
-| (lang dune 2.7)
-Done: 8/11 (jobs: 1)
 ```
 
-When we change our program, we type `dune build` again to make a new
-executable. We can run the executable with `dune exec` (it's called
-`helloworld.exe` even when we're not using Windows):
+When we change our program, we can type `dune build` again to make a new
+executable. To run the program, we can use
 
 ```
-$ dune exec ./helloworld.exe
-Hello, World!        
+$ dune exec ./bin/main.exe
+Hello, World!
 ```
 
-Let's look at the contents of our new directory. Dune has added the
-`helloworld.ml` file, which is our OCaml program. It has also added our `dune`
-file, which tells dune how to build the program, and a `_build` subdirectory,
-which is dune's working space.
+Or, alternatively,
 
 ```
-$ ls
-_build		dune		helloworld.ml
+$ dune exec hellworld
+Hello, World!
 ```
```

Comment:
I'm not convinced by this "alternatively", because I think it is too magic (where does the mapping from `helloworld` to `./bin/main.exe` come from? Does Dune implement the convention that `<dir>/bin/main.ml` are also runnable as `dune exec <dir>`?)

Reply:
My own feeling is that this command is much more user friendly, especially for people just getting started. Executing the program by giving a path to *a file that doesn't exist* is quite strange, in my experience, and iirc, I've seen new learners get thrown off by it. However, other tool chains that users are likely familiar with (e.g., npm, cargo, sbt, etc.) all have this convention of being able to run the executable under development by name.

I would ask too, why is running an executable by name (which is what anyone who uses terminal shell's is used to any how) more magical than running an executable by a non-existent path? :)

Note that this point is not altogether unrelated to correcting the command: previously, the generated file had the same name as the executable's `public_name` so there was somewhat less room for confusion.

What do you think?  Do you find my reasoning here compelling? Should we solicit input (perhaps from some people who are just learning OCaml)? Or are you firm on your position and I should just remove this part?

Reply:
I think that your experience is at least as valid as mine (you know the dune UI better and possibly have had more recent interactions with dune beginners), and in addition you are the one who wrote the PR, so I would suggest to follow your gut feeling here instead of mine.